samples/tools/autogenbench/scenarios/GAIA/Scripts/demystifying-telemetry-sqlite.ipynb

```diff
@@ -0,0 +1,700 @@
+    "    if index[1] > 0:\n",
+    "        index[1] -= 1\n",
+    "        update_content(outer_event_index=index[0], inner_event_index=index[1])\n",
+    "    elif index[0] > 0:\n",
+    "        index[0] -= 1\n",
+    "        index[1] = len(data[index[0]][\"inner\"]) - 1\n",
+    "        update_content(outer_event_index=index[0], inner_event_index=index[1])\n",
+    "\n",
+    "\n",
+    "def on_next_button_clicked(b):\n",
+    "    if index[1] < len(data[index[0]][\"inner\"]) - 1:\n",
+    "        index[1] += 1\n",
+    "        update_content(outer_event_index=index[0], inner_event_index=index[1])\n",
+    "    elif index[0] < len(data) - 1:\n",
+    "        index[0] += 1\n",
+    "        index[1] = 0\n",
+    "        update_content(outer_event_index=index[0], inner_event_index=index[1])\n",
+    "\n",
+    "\n",
+    "def on_prev_phase_button_clicked(b):\n",
+    "    if index[0] > 0:\n",
+    "        index[0] -= 1\n",
+    "        index[1] = 0\n",
+    "        update_content(outer_event_index=index[0], inner_event_index=index[1])\n",
+    "\n",
+    "\n",
+    "def on_next_phase_button_clicked(b):\n",
+    "    if index[0] < len(data) - 1:\n",
+    "        index[0] += 1\n",
+    "        index[1] = 0\n",
+    "        update_content(outer_event_index=index[0], inner_event_index=index[1])\n",
+    "\n",
+    "\n",
+    "prev_button.on_click(on_prev_button_clicked)\n",
+    "next_button.on_click(on_next_button_clicked)\n",
+    "prev_phase_button.on_click(on_prev_phase_button_clicked)\n",
+    "next_phase_button.on_click(on_next_phase_button_clicked)\n",
+    "\n",
+    "button_grid = widgets.GridBox(\n",
+    "    children=[prev_phase_button, prev_button, next_phase_button, next_button],\n",
+    "    layout=widgets.Layout(grid_template_rows=\"auto auto\", grid_template_columns=\"49% 49%\", grid_gap=\"5px 5px\"),\n",
+    ")\n",
+    "\n",
+    "container_layout = widgets.Layout(width=\"820px\")\n",
+    "container = widgets.VBox(\n",
+    "    [\n",
+    "        input_box,\n",
+    "        status_msg,\n",
+    "        button_grid,\n",
+    "        outer_event_textarea,\n",
+    "        event_with_image_textarea,\n",
+    "        events_without_image_textarea,\n",
+    "        image_widget,\n",
+    "    ],\n",
+    "    layout=container_layout,\n",
+    ")\n",
+    "\n",
+    "display(container)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## EXTRA: Diff between states - in this case cookie states\n",
+    "\n",
+    "We could also figure out what 'json_state' has changed in between same event types. In this section we try to debug how and when the cookie state changes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from jsondiff import diff\n",
+    "\n",
+    "final = merged_and_nested\n",
+    "\n",
+    "cookies_events = [event for event in final if event[\"event_name\"] == \"oookies\"]\n",
```

Comment:
I'm getting KeyError: 'event_name' from this line

Reply:
fixed thanks